src/Dojo.Net/BaseClient.cs

```diff
@@ -26,8 +26,10 @@ public class BaseClient
         /// <param name="clientAuthorization">The IClientAuthorization instance</param>
         public BaseClient(HttpClient client, IClientAuthorization clientAuthorization)
         {
-            _client = client ?? throw new ArgumentNullException(nameof(client));
-            _clientAuthorization = clientAuthorization ?? throw new ArgumentNullException(nameof(clientAuthorization));
+            Client = client ?? throw new ArgumentNullException(nameof(client));
+            _ = clientAuthorization ?? throw new ArgumentNullException(nameof(clientAuthorization));
+            clientAuthorization.AuthorizeRequestsAsync(client);
```

Comment:
Are you sure it is a good idea to make non awaitable call in the constructor?

Reply:
I will rectify, that jould not be an async in the first place. Good spot.